src/openslide-vendor-generic-tiff.c

```diff
@@ -175,6 +175,20 @@ static int width_compare(gconstpointer a, gconstpointer b) {
   }
 }
 
+static void set_resolution_prop(openslide_t *osr, TIFF *tiff,
```

Comment:
Rather than duplicating the code from Leica, let's move this into the generic TIFF decoder and have both drivers call it.  In cases where either `tiff` or `tifflike` could provide some functionality, we usually put it into `tifflike`, so let's make it a helper function there.  But for now I'd say let's _not_ make it happen automatically from `store_and_hash_properties`, since there are lots of formats that record the MPP in unusual ways, and it's better for drivers to configure the MPP explicitly.